runtime/common/src/assigned_slots.rs

```diff
@@ -0,0 +1,1281 @@
+	/// but not earlier than the specified period.
+	pub period_begin: LeasePeriod,
+	/// Number of lease period the slot lease will last.
+	/// This is set to the value configured in `TemporarySlotLeasePeriodLength`.
+	pub period_count: LeasePeriod,
+	/// Last lease period this slot had a turn in (incl. current).
+	/// This is set to the beginning period of a slot.
+	pub last_lease: Option<LeasePeriod>,
+	/// Number of leases this temporary slot had (incl. current).
+	pub lease_count: u32,
+}
+
+type BalanceOf<T> = <<<T as Config>::Leaser as Leaser<<T as frame_system::Config>::BlockNumber>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+type LeasePeriodOf<T> =
+	<<T as Config>::Leaser as Leaser<<T as frame_system::Config>::BlockNumber>>::LeasePeriod;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::generate_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	#[pallet::disable_frame_system_supertrait_check]
+	pub trait Config: configuration::Config + paras::Config + slots::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Origin for assigning slots.
+		type AssignSlotOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+
+		/// The type representing the leasing system.
+		type Leaser: Leaser<
+			Self::BlockNumber,
+			AccountId = Self::AccountId,
+			LeasePeriod = Self::BlockNumber,
+		>;
+
+		/// The number of lease periods a permanent parachain slot lasts.
+		#[pallet::constant]
+		type PermanentSlotLeasePeriodLength: Get<u32>;
+
+		/// The number of lease periods a temporary parachain slot lasts.
+		#[pallet::constant]
+		type TemporarySlotLeasePeriodLength: Get<u32>;
+
+		/// The max number of permanent slots that can be assigned.
+		#[pallet::constant]
+		type MaxPermanentSlots: Get<u32>;
+
+		/// The max number of temporary slots that can be assigned.
+		#[pallet::constant]
+		type MaxTemporarySlots: Get<u32>;
+
+		/// The max number of temporary slots to be scheduled per lease periods.
+		#[pallet::constant]
+		type MaxTemporarySlotPerLeasePeriod: Get<u32>;
+	}
+
+	/// Assigned permanent slots, with their start lease period.
+	#[pallet::storage]
+	#[pallet::getter(fn permanent_slots)]
+	pub type PermanentSlots<T: Config> =
+		StorageMap<_, Twox64Concat, ParaId, Option<LeasePeriodOf<T>>, ValueQuery>;
+
+	/// Number of assigned (and active) permanent slots.
+	#[pallet::storage]
+	#[pallet::getter(fn permanent_slot_count)]
+	pub type PermanentSlotCount<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	/// Assigned temporary slots.
+	#[pallet::storage]
+	#[pallet::getter(fn temporary_slots)]
+	pub type TemporarySlots<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ParaId,
+		Option<ParachainTemporarySlot<T::AccountId, LeasePeriodOf<T>>>,
+		ValueQuery,
```

Comment:
This doesn't make sense as is.

```suggestion
		ParachainTemporarySlot<T::AccountId, LeasePeriodOf<T>>,
		OptionQuery,
```